extensions/blocks/podcast-player/edit.js

```diff
@@ -35,14 +41,53 @@ const handleSSRError = () => {
 	return <p>{ __( 'Failed to load Block', 'jetpack' ) }</p>;
 };
 
+// Support page link.
+const supportLink =
+	isSimpleSite() || isAtomicSite()
+		? 'http://en.support.wordpress.com/wordpress-editor/blocks/podcast-player-block/'
+		: 'https://jetpack.com/support/jetpack-blocks/podcast-player-block/';
+
+/**
+ * Control component used either inside of the Podcast block
+ * as well as into the InspectorControls section (sidebar)
+ *
+ * @param {string}      url             Podcast Feed URL
+ * @param {string|Null} label           Input field label. Optional.
+ * @param {Function}    onUrlChange     onChange text handler.
+ * @param {Function}    onEnterKeyPress on Enter key input event handler. Optional.
+ * @param {Function}    onBlur          onBlur input handler. Optional.
+ * @return {*}                          React component
+ */
+const FeedURLControl = ( {
+	url,
+	label = null,
+	onUrlChange,
+	onEnterKeyPress = noop,
+	onBlur = noop,
+} ) => (
+	<TextControl
+		label={ label }
+		type="url"
+		placeholder={ __( 'Enter URL here…', 'jetpack' ) }
+		value={ url || '' }
+		className={ 'components-placeholder__input' }
+		onChange={ onUrlChange }
+		onKeyPress={ ev => ( ev.key === 'Enter' ? onEnterKeyPress( ev ) : noop ) }
+		onBlur={ onBlur }
+	/>
+);
```

Comment:
I'm not sure I'm too big of a fan of this abstraction. It doesn't seem to save us much work and obfuscates the underlying component. What do you think?

Reply:
I'm interested to understand _why_ this abstraction was added. At first glance I can't see any major benefit but @retrofox may well have a good reason that we're not aware of.

Reply:
Looking at the rest of the changes I support this component abstraction. You will notice that `FeedURLControl` is used twice in the component

1. Block Controls
2. Inspector Controls

It makes sense to DRY this up especially given the number of props being applied. It would be easier to change one and forget the other. Having a single component helps avoid this.

Reply:
> I'm not sure I'm too big of a fan of this abstraction.

It's part of the implementation. The idea is to reuse the same component in two places.

>  It doesn't seem to save us much work and obfuscates the underlying component. What do you think?

Disagree. ;-)
The component takes over updating the Podcast Link. It expects three arguments and 💥 you get the input filed ready to use. Glad to update the doc if we consider appropriate.

> Looking at the rest of the changes I support this component abstraction. You will notice that FeedURLControl is used twice in the component

It's already documented in the function :-D :

```es6
/**
 * Control component used either inside of the Podcast block
 * as well as into the InspectorControls section (sidebar)
 *
 * @param {string}      url             Podcast Feed URL
 * @param {string|Null} label           Input field label. Optional.
 * @param {Function}    onUrlChange     onChange text handler.
 * @param {Function}    onEnterKeyPress on Enter key input event handler. Optional.
 * @param {Function}    onBlur          onBlur input handler. Optional.
 * @return {*}                          React component
 */
```

Maybe it' unclear or incomplete. Happy to improve it :-) 

> It makes sense to DRY this up especially given the number of props being applied. 

Agree.

Thanks, @obenland and @getdave for this feedback. My suggestion is keeping this component as it, maybe improving the jsdoc. But glad to revert it if we consider good, and _kind of_ duplicate the code un both places where the component is consumed.